### PR TITLE
[Link] Documentation and tester update for link on Android

### PR DESCRIPTION
**Pull request by rohanprasadofficial**
### Platforms Impacted
- [ ] iOS
- [ ] macOS
- [ ] win32 (Office)
- [ ] windows
- [x] android

### Description of changes
Updates the documentation and tester app for Link for android. 

### Verification
Visual Verification on Android

### Pull request checklist

This PR has considered (when applicable):
- [ ] Automated Tests
- [x] Documentation and examples
- [ ] Keyboard Accessibility
- [ ] Voiceover
- [ ] Internationalization and Right-to-left Layouts

### CODE REVIEW

File: packages/experimental/Link/src/Link.types.ts [hunk 1]
```diff
@@ -7,6 +7,7 @@ export const linkName = 'Link';
 export type LinkState = IPressableState & {
   /**
    * Specifies whether the link has been visited.
+   * Note: Not supported for Android
```
Comment (rurikoaraki) [inline packages/experimental/Link/src/Link.types.ts:10]: Please update similar comments in SPEC.md

Comment (rohanprasadofficial) [reply packages/experimental/Link/src/Link.types.ts:10]: Seems in the experimental folder we don't have a SPEC.md for Link.
But the SPEC.md is present in the components folder

Comment (rurikoaraki) [reply packages/experimental/Link/src/Link.types.ts:10]: Yes, I moved the SPEC file there to prepare LinkV1 for release.

Comment (rurikoaraki) [reply packages/experimental/Link/src/Link.types.ts:10]: Once this is fixed feel free to merge

Comment (rohanprasadofficial) [reply packages/experimental/Link/src/Link.types.ts:10]: Updated spec.md.

**Pull request merged**